### PR TITLE
Setup test infrastructure and add a simple test

### PR DESCRIPTION
**Pull request by ricrogz**
I have set up the infrastructure to build a test executable, and added a simple test as example.

Also, to make it easier to add tests, I have extracted and exported the simple maeparser `Block` to `sketcherMinimizerMolecule` parser that is used to read the templates used the minimizer.

When I added the test, I found a precision/rounding issue that caused some setups to produce very different results than others. It all started with a minimal difference in the score produced by `sketcherMinimizer::testAlignment()`, which caused an extra rotation inversion. I patched this adding some rounding to improve reproducibility of the results.

As a side node, I have noticed a great deal of arguments passed by value; it might be worth taking a look at them and turn them into references.

### CODE REVIEW

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -1,2 +1,15 @@
 # Use the example executable as a test
-add_test(example ${CMAKE_BINARY_DIR}/example_dir/example)
\ No newline at end of file
+add_test(example ${CMAKE_BINARY_DIR}/example_dir/example)
+
+# Require Boost testing framework for real tests
```
Comment (d-b-w) [inline test/CMakeLists.txt]: What does "real" mean in this sentence?

File: test/test_coordgen.cpp [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    auto block = r.next(mae::CT_BLOCK);
+    BOOST_REQUIRE(block != nullptr);
+
+    auto* mol = mol_from_mae_block(*block);
+    BOOST_REQUIRE(mol != nullptr);
+    BOOST_REQUIRE_EQUAL(mol->getAtoms().size(), 26);
+    BOOST_REQUIRE_EQUAL(mol->getBonds().size(), 26);
+
+    sketcherMinimizer minimizer;
+    minimizer.initialize(mol); // minimizer takes ownership of mol
```
Comment (d-b-w) [inline test/test_coordgen.cpp]: Thanks for the comment.

File: test/CMakeLists.txt (hunk 1 above)
Comment (ricrogz) [reply test/CMakeLists.txt]: Up to this point, we had the example as the only test in this repo, and that does not look like a real test, nothing is checked.

File: test/test_coordgen.cpp (hunk 2 above)
Comment (ricrogz) [reply test/test_coordgen.cpp]: No problem. I was quite puzzled when I attempted to clean up mol and valgrind returned an error. Then I found out that the minimizer takes ownership and cleans it up on destruction.

File: test/CMakeLists.txt (hunk 1 above)
Comment (d-b-w) [reply test/CMakeLists.txt]: ahaha.

**Pull request merged**